Feat: added global loading state and moved loading indicator to central location

john-salman opened the pull request:
### Description:
Added a loading bar to the bottom of the map and table

### To Test:
1. Hover a mainstem in the search results, loading bar should show
2. Select a mainstem in the search results, loading bar should show
3. Repeat the above steps for table and map
4. Select a mainstem on the map, loading bar should show

webb-ben commented on src/app/features/Loading.tsx:
```diff
@@ -0,0 +1,17 @@
+    return (
+        <>
+            {loading && (
+                <div id="loading-bar" className="absolute bottom-0 h-1 w-full">
```
Might be worth bumping the height ever so slightly

C-Loftus commented:
I am seeing some bodies of water in which the loader loads indefinitely and never stops. Feel free to call me and I can share screen visually. I think this might be because there is no data with them? I could be wrong.

this one below for instance 

![image](https://github.com/user-attachments/assets/0148218c-8bdf-458d-b1fb-c5ae57169b29)

C-Loftus commented:
I also sort of think the loader icon could be moved to the top. I feel like I don't usually see loaders at the bottom of the page but thats just me.

I agree with ben to increase the height and make it more visually prominent.

webb-ben commented:
> I am seeing some bodies of water in which the loader loads indefinitely and never stops. Feel free to call me and I can share screen visually. I think this might be because there is no data with them? I could be wrong.
> 
> this one below for instance
> 
> ![image](https://private-user-images.githubusercontent.com/70598503/427056066-0148218c-8bdf-458d-b1fb-c5ae57169b29.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI5OTc5ODMsIm5iZiI6MTc0Mjk5NzY4MywicGF0aCI6Ii83MDU5ODUwMy80MjcwNTYwNjYtMDE0ODIxOGMtOGJkZi00NThkLWIxZmItYzVhZTU3MTY5YjI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI2VDE0MDEyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg5M2Q1MjRmZGM5YTQ5YzgyMmQwNDFjMmZlZGUzOTkyM2EzZTZjNzEyZDY1NWQyYzBhNjgzYjE3MmM1MGRmZDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0LAsqHR_PmQ2EuJmztL50mFpDuHm6BFmn5XwhdPdsmg)

This has no datasets in the graph. See https://reference.geoconnex.dev/collections/mainstems/items/525564

C-Loftus commented:
One other comment, it seems like while the map is panning to the location, it will say there are no datasets instead of having some sort of loading state or loading async (showing datasets there in parallel while the map updates)

See the charles river, which has datasets and is loading in, but until it is fully loaded, says it has no datasets
![image](https://github.com/user-attachments/assets/f7146fd3-d4dc-469f-93d6-86e65f2364e6)

john-salman commented:
> See the charles river, which has datasets and is loading in, but until it is fully loaded, says it has no datasets

Added a special case for this.

Addressed the other comments and moved the no-datasets handling from #19 into this PR  as well
